core/services/workflows/v2/engine.go

```diff
@@ -307,6 +314,9 @@ func (e *Engine) startExecution(ctx context.Context, wrappedTriggerEvent enqueue
 	}
 
 	// TODO(CAPPL-736): handle execution result
+
+	e.meterReports.Delete(executionID)
```

Comment:
We probably want to call Delete() also if we return earlier on error?

Reply:
Yes, good catch

Reply:
Maybe just put the Delete() before err check? Then you can have a single one.